### DISCUSSION TITLE
Call validation for alpine?

### DISCUSSION THREAD

**Discussion by QJan84** [alpinejs/alpine#523] [4. General]
Hi!

I have problems with the order of my functions.
`data-trigger="validate"` starts the validation.

`component()` is executed first and then the validation script. This is not a good thing because I query the **.is--valid** class and then send the form.

How can I first run validation and then `submitForm()`?

Here is my example with a form in a modal:

HTML:

		<div x-data="component()" x-init="init()">
			<form x-ref="form" x-show="!status" data-trigger="validate">

				...form data
				...form data
				...form data

				<button type="submit" @click="submitForm($refs)">
					Submit
				</button>

			</form>
		</div>

JS:

	function component () {
		return {
			callbackModal: false,
			status: false,
			init(e) {
				var myForm = this.$refs.form;
				// Prevents the form from being sent in the validation script.
				myForm.classList.add("is--fetch");
			},
			submitForm() {
				if(this.$refs.form.classList.contains('is--valid')){
					fetch('/php/ajax/send_callback.php', {
						method: 'POST',
						body: new FormData(this.$refs.form)
					})
					.then(response => response.ok && response.json()).then(data => {
						this.status = data.status // Show valid or error message
					});
				}
			}
		}
	}

**Comment by HugoDF** [comment]
I'm not sure how `data-validate` works but you'll need to call whatever library is doing the validation in submitForm and based on that, submit or bail.
